lib/spack/spack/util/environment.py

```diff
@@ -1048,11 +1048,12 @@ def _source_single_file(file_and_args, environment):
             [source_file, suppress_output, concatenate_on_success, dump_environment_cmd]
         )
 
+        # List fails to preserve nested double quotes on Windows
```

Comment:
Lists do not modify their contents implicitly, the quote stripping is due to CMD's handling of quotes and the process of passing these strings from python to the system call. Please update the comment to something like "Passing this expression to CMD requires the use of Popen's string based args to avoid unexpected parsing by the underlying subprocess module https://docs.python.org/3.8/library/subprocess.html#notes"